src/Object/Post.php

```diff
@@ -82,8 +82,9 @@ public function __construct(array $data)
 					continue;
 				}
 
-				// You can always comment on Diaspora items
-				if (($item['network'] == NETWORK_DIASPORA) && (local_user() == $item['uid'])) {
+				// You can always comment on Diaspora and OStatus items
+				// The empty network can happen with your local (self) contact
+				if (in_array($item['network'], ['', NETWORK_OSTATUS, NETWORK_DIASPORA]) && (local_user() == $item['uid'])) {
```

Comment:
Is there any way to ensure the `'network'` value is among the available ones?

Reply:
I don't understand?

Reply:
`$item['nework']` should always be amongst the expected values, can we ensure this before this check?

Reply:
This is more or less a bugfixing. Normally the "writable" value is defined by the contact. On Twitter and other external networks this value was always set to "true" (so no need to change here). But for Diaspora this wasn't the case - although technically you could always comment. So we just fix it here.

And with the empty network we just correct the problem that the "self" contact hasn't set this value as well.

This is some temporarily solution and I guess we will eventually find a better way.

Reply:
To be clear: We only **overwrite** the contact value.

Reply:
Ok, as long as you keep it in mind.

Reply:
You can be sure of this. Just think of "last-child". This is something similar that I have to address during the item structure reworking.